imports/scaleform/client.lua

```diff
@@ -0,0 +1,231 @@
+---@class renderTargetTable
+---@field name string
+---@field model string|number
+
+---@class detailsTable
+---@field name string
+---@field fullScreen? boolean
+---@field x? number
+---@field y? number
+---@field width? number
+---@field height? number
+---@field renderTarget? renderTargetTable
+
+---@class Scaleform : OxClass
+---@field scaleform number
+---@field draw boolean
+---@field target number
+---@field targetName string
+---@field handle number
+---@field fullScreen boolean
+lib.scaleform = lib.class('Scaleform')
+
+--- Converts the arguments into data types usable by scaleform
+---@param argsTable table
+local function convertArgs(argsTable)
+    for i = 1, #argsTable do
+        local arg = argsTable[i]
+        local argType = type(arg)
+
+        if argType == 'string' then
+            ScaleformMovieMethodAddParamPlayerNameString(arg)
+        elseif argType == 'number' then
+            if math.type(arg) == 'integer' then
+                ScaleformMovieMethodAddParamInt(arg)
+            else
+                ScaleformMovieMethodAddParamFloat(arg)
+            end
+        elseif argType == 'boolean' then
+            ScaleformMovieMethodAddParamBool(arg)
+        else
+            error(('Unsupported Parameter type [%s]'):format(argType))
+        end
+    end
+end
+
+---@param expectedType 'boolean' | 'integer' | 'string'
+---@return boolean | integer | string
+---@description Awaits the return value, and converts it to a usable data type
+local function retrieveReturnValue(expectedType)
+    local result = EndScaleformMovieMethodReturnValue()
+
+    lib.waitFor(function()
+        if IsScaleformMovieMethodReturnValueReady(result) then
+            return true
+        end
+    end, "Failed to retrieve return value", 1000)
+
+    if expectedType == "integer" then
+        return GetScaleformMovieMethodReturnValueInt(result)
+    elseif expectedType == "boolean" then
+        return GetScaleformMovieMethodReturnValueBool(result)
+    else
+        return GetScaleformMovieMethodReturnValueString(result)
+    end
+end
+
+---@param details detailsTable | string
+---@return nil
+---@description Create a new scaleform class
+function lib.scaleform:constructor(details)
+    details = type(details) == "table" and details or { name = details }
+
+    local scaleform = lib.requestScaleformMovie(details.name)
+    if not scaleform then
+        return error(('Failed to request scaleform movie - [%s]'):format(details.name))
+    end
+
+    self.handle = scaleform
+    self.draw = false
+
+    self.fullScreen = details.fullScreen ~= nil and details.fullScreen or true
+    self.x = details.x or 0
+    self.y = details.y or 0
+    self.width = details.width or 0
+    self.height = details.height or 0
+
+    if details.renderTarget then
+        self:setRenderTarget(details.renderTarget.name, details.renderTarget.model)
+    end
+end
+
+---@param name string
+---@param args? table
```

Comment:
Should not be marked as optional, as your type check ensures it is a table. You should also clarify that it's an array-like table and which types are valid - `(number | string | boolean)[]`